src/System.Linq.Parallel/src/System/Linq/Parallel/Enumerables/ParallelQuery.cs

```diff
@@ -102,7 +102,7 @@ internal ParallelQuery(QuerySettings settings)
 
         internal sealed override ParallelQuery<TCastTo> Cast<TCastTo>()
         {
-            return ParallelEnumerable.Select<TSource, TCastTo>(this, elem => (TCastTo)(object)elem);
+            return ParallelEnumerable.Select<TSource, TCastTo>(this, elem => (TCastTo)(object)elem!);
```

Comment:
Can it instead be casted  to `object?` (`(TCastTo)(object?)elem`

Reply:
yes it can cast to `(object?)` but cannot cast from there to `(TCastTo)`, so even i put object casting nullable still need to bang for `(TCastTo)` cast .